pkg/action/archive.go

```diff
@@ -33,25 +36,27 @@ func copyArchive(src string, dst string) error {
 
 	defer func() {
 		if cerr := w.Close(); cerr != nil {
-			log.Printf("failed to close file: %v", cerr)
+			err = fmt.Errorf("failed to close file: %v", cerr)
```

Comment:
love the idea, and actually was thinking about proposing this earlier, but there's plenty of other places where we don't actually deal with errors, so didn't want to make you jump through the hoops.

I think (and I could be totally wrong here...) You would have to make the return `error` be a named return argument, because this defer will run after the ones below and will have no effect on setting it. So, I think to make this return in case of close failure, we'd have to:
 * change the signature to return (err error)
 * but then the problem is that if the L44 returns an error, this defer might fail and if it does it will overwrite the error
 * so maybe use something like errors.Join (which will ignore nil values, so I think it should "just work").

Here's one article I found:
https://riandyrn.medium.com/golang-magic-modify-return-value-using-deferred-function-ed0eabdaa75

After having said all though, since we don't check the error on r.Close on L30, I think just logging this as an error here will be fine.

Reply:
Makes sense! I'll just log the error for now.

Reply:
Changed in [`be839e6` (#144)](https://github.com/chainguard-dev/bincapz/pull/144/commits/be839e6d7ef7147bc93db2321cf097e64f2c4e5f).